### PR TITLE
Compiling for HaikuOS 64 bits 

### PR DESCRIPTION
**Pull request by Sylvain78**
configure files to recognize the HaikuOS platform on x86/amd 64 bits

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -1047,6 +1047,8 @@ AS_CASE([$host],
     [arch=i386; system=bsd_elf],
   [[i[3456]86-*-haiku*]],
     [arch=i386; system=beos],
+  [x86_64-unknown-haiku],
```
Comment (dra27) [inline configure.ac]:
```suggestion
  [x86_64-*-haiku*],
```

**Review by dra27 — COMMENTED** [review]
Seems fine to me - I'd update the regexp to be similar to the 32-bit version.

The order of those arches is already a little inconsistent, but please could you move it to be the with the other x86_64 cases around L1101 (only asking as the regexp wants tweaking!)

**Comment by Sylvain78** [comment]
Changes applied as requested

**Comment by dra27** [comment]
Thank you! Seems to be working fine on a quickly fired up VM, apart from a few failing tests!

**Pull request merged**